pkg/controller/zerocapacity/zero_controller.go

```diff
@@ -419,17 +398,14 @@ func (z *Controller) zeroPodSpec(name, namespace string, configMap *corev1.Confi
 	}
 
 	if zeroSpec.Volume.UpdatePermissions {
-		pod.Spec.InitContainers = []corev1.Container{
-			ispnCtrl.ChmodInitContainer("backup-chmod-pv", name, zeroSpec.Volume.MountPath),
-		}
+		ispnCtrl.AddVolumeChmodInitContainer("backup-chmod-pv", name, zeroSpec.Volume.MountPath, &pod.Spec)
 	}
 
-	ispnCtrl.AddServerDataChmodInitContainer(dataVolName, &pod.Spec)
```

Comment:
@ryanemerson do we really need to add chmod init container for zero pod data volume as it's EmptyDir, not real volume
Also OCP tests were fails as backup-chmod-pv adds without lookup MAKE_DATADIR_WRITABLE env variable.
wdyt about this?

Reply:
> do we really need to add chmod init container for zero pod data volume as it's EmptyDir, not real volume

Possibly not. I made this change when I was investigating the CI failures, so I think we can remove it.



> Also OCP tests were fails as backup-chmod-pv adds without lookup MAKE_DATADIR_WRITABLE env variable.
> wdyt about this?

Right, makes sense to check the env var then. I think minikube will fail without the initContainer, but I can just set `MAKE_DATADIR_WRITABLE` in that case.

Reply:
Thanks, already fixed.